app/models/concerns/alegre_webhooks.rb

```diff
@@ -13,11 +13,17 @@ def valid_request?(request)
     def webhook(request)
       begin
         doc_id = request.params.dig('data', 'requested', 'id')
+        doc_id = request.params.dig('data', 'item', 'id') if doc_id.nil?
+        is_from_alegre_callback = request.params.dig('data', 'item', 'callback_url').to_s.include?("/presto/receive/add_item")
         raise 'Unexpected params format' if doc_id.blank?
-        redis = Redis.new(REDIS_CONFIG)
-        key = "alegre:webhook:#{doc_id}"
-        redis.lpush(key, request.params.to_json)
-        redis.expire(key, 1.day.to_i)
+        if is_from_alegre_callback
+          Bot::Alegre.process_alegre_callback(request.params)
+        else
+          redis = Redis.new(REDIS_CONFIG)
+          key = "alegre:webhook:#{doc_id}"
+          redis.lpush(key, request.params.to_json)
+          redis.expire(key, 1.day.to_i)
```

Comment:
I suggest we add the `redis.expire` call under an `ensure` block... this way, even if something goes wrong, we make sure that Redis is going to be cleared (in that case we may need to declare `redis` outside the begin/rescue block).